mmaction/datasets/pipelines/augmentations.py

```diff
@@ -1448,183 +1449,98 @@ def __repr__(self):
-        """
-        if np.random.rand() > 0.5:
-            u = np.cos(alpha * np.pi)
-            w = np.sin(alpha * np.pi)
-            bt = np.array([[1.0, 0.0, 0.0], [0.0, u, -w], [0.0, w, u]],
-                          dtype=np.float32)
-            tyiq = np.array([[0.299, 0.587, 0.114], [0.596, -0.274, -0.321],
-                             [0.211, -0.523, 0.311]],
-                            dtype=np.float32)
-            ityiq = np.array([[1.0, 0.956, 0.621], [1.0, -0.272, -0.647],
-                              [1.0, -1.107, 1.705]],
-                             dtype=np.float32)
-            t = np.dot(np.dot(ityiq, bt), tyiq).T
-            t = np.array(t, dtype=np.float32)
-            img = np.dot(img, t)
-        return img
+    @staticmethod
+    def adjust_hue(img, factor):
+        img = np.clip(img, 0, 255).astype(np.uint8)
+        hsv = cv2.cvtColor(img, cv2.COLOR_RGB2HSV)
+        offset = int(factor * 255)
+        hsv[..., 0] = (hsv[..., 0] + offset) % 180
+        img = cv2.cvtColor(hsv, cv2.COLOR_HSV2RGB)
+        return img.astype(np.float32)
+
+    def __init__(self, brightness=0.5, contrast=0.5, saturation=0.5, hue=0.1):
+        self.brightness = self.check_input(brightness, 1, 1)
+        self.contrast = self.check_input(contrast, 1, 1)
+        self.saturation = self.check_input(saturation, 1, 1)
+        self.hue = self.check_input(hue, 0.5, 0)
+        self.fn_idx = np.random.permutation(4)
 
     def __call__(self, results):
         imgs = results['imgs']
-        out = []
-        if self.color_space_aug:
-            bright_delta = np.random.uniform(-32, 32)
-            contrast_alpha = np.random.uniform(0.6, 1.4)
-            saturation_alpha = np.random.uniform(0.6, 1.4)
-            hue_alpha = np.random.uniform(-18, 18)
-            jitter_coin = np.random.rand()
-            for img in imgs:
-                img = self.brightness(img, delta=bright_delta)
-                if jitter_coin > 0.5:
-                    img = self.contrast(img, alpha=contrast_alpha)
-                    img = self.saturation(img, alpha=saturation_alpha)
-                    img = self.hue(img, alpha=hue_alpha)
-                else:
-                    img = self.saturation(img, alpha=saturation_alpha)
-                    img = self.hue(img, alpha=hue_alpha)
-                    img = self.contrast(img, alpha=contrast_alpha)
-                out.append(img)
-        else:
-            out = imgs
-
-        # Add PCA based noise
-        alpha = np.random.normal(0, self.alpha_std, size=(3, ))
-        rgb = np.array(
-            np.dot(self.eig_vec * alpha, self.eig_val), dtype=np.float32)
-        rgb = rgb[None, None, ...]
-
-        results['imgs'] = [img + rgb for img in out]
-        results['eig_val'] = self.eig_val
-        results['eig_vec'] = self.eig_vec
-        results['alpha_std'] = self.alpha_std
-        results['color_space_aug'] = self.color_space_aug
-
+        num_clips, clip_len = 1, len(imgs)
```

Comment:
`num_clips` is hardcoded to 1 ?

Reply:
This `num_clips` is only used in the ColorJitter. Here it indicates we perform the same augmentation for all frames (the same brightness, contrast, saturation, hue factor). If set to the real num_clips, it indicates we perform different augmentations for different clips (in case we sample multiple clips from a video).

Reply:
We don't sample multiple clips in training for recognition. However, such operation is needed in self-supervised pretraining, which may be added in the future.